test/client-bulk.test.ts

```diff
@@ -0,0 +1,236 @@
+import { it, expect } from "vitest";
+import * as msw from "msw";
+import * as assert from "assert";
+
+import { createClient } from "./__testutils__/createClient";
+import { isAuthorizedRequest } from "./__testutils__/isAuthorizedRequest";
+import { testFetchOptions } from "./__testutils__/testFetchOptions";
+
+import * as prismicCustomTypes from "../src";
+
+it("performs a bulk transaction", async (ctx) => {
+	const insertedCustomType = ctx.mock.model.customType();
+	const updatedCustomType = ctx.mock.model.customType();
+	const deletedCustomType = ctx.mock.model.customType();
+
+	const insertedSlice = ctx.mock.model.sharedSlice();
+	const updatedSlice = ctx.mock.model.sharedSlice();
+	const deletedSlice = ctx.mock.model.sharedSlice();
+
+	const operations = [
+		{
+			type: prismicCustomTypes.BulkOperationType.CustomTypeInsert,
+			id: insertedCustomType.id,
+			payload: insertedCustomType,
+		},
+		{
+			type: prismicCustomTypes.BulkOperationType.CustomTypeUpdate,
+			id: updatedCustomType.id,
+			payload: updatedCustomType,
+		},
+		{
+			type: prismicCustomTypes.BulkOperationType.CustomTypeDelete,
+			id: deletedCustomType.id,
+			payload: { id: deletedCustomType.id },
+		},
+		{
+			type: prismicCustomTypes.BulkOperationType.SliceInsert,
+			id: insertedSlice.id,
+			payload: insertedSlice,
+		},
+		{
+			type: prismicCustomTypes.BulkOperationType.SliceUpdate,
+			id: updatedSlice.id,
+			payload: updatedSlice,
+		},
+		{
+			type: prismicCustomTypes.BulkOperationType.SliceDelete,
+			id: deletedSlice.id,
+			payload: { id: deletedSlice.id },
+		},
+	];
+
+	const client = createClient(ctx);
+
+	ctx.server.use(
+		msw.rest.post(
+			new URL("./bulk", client.endpoint).toString(),
+			async (req, res, ctx) => {
+				if (!isAuthorizedRequest(client, req)) {
+					return res(
+						ctx.status(403),
+						ctx.json({ message: "[MOCK FORBIDDEN ERROR]" }),
+					);
+				}
+
+				assert.deepStrictEqual(await req.json(), {
```

Comment:
It behaves the same, but the API and output is slightly different.

I opted for `assert` because `expect` didn't seem to throw in a MSW handler the same way `assert`, but it seems to work fine now. 🤷‍♂️

I changed all uses of `assert` to `expect`.